configuration/ibm/50001002.json

```diff
@@ -0,0 +1,3196 @@
+                    "xyz.openbmc_project.Inventory.Connector.Port": null,
+                    "com.ibm.ipzvpd.Location": {
+                        "LocationCode": "Ufcs-P0-C10-T1"
+                    },
+                    "xyz.openbmc_project.Inventory.Item": {
+                        "PrettyName": "NVMe Internal Connector"
+                    }
+                }
+            },
+            {
+                "inventoryPath": "/xyz/openbmc_project/inventory/system/chassis/motherboard/pcieslot10/pcie_card10/c10_connector2",
+                "serviceName": "xyz.openbmc_project.Inventory.Manager",
+                "inherit": false,
+                "noprime": true,
+                "ccin": ["6B87"],
+                "extraInterfaces": {
+                    "xyz.openbmc_project.Inventory.Item.Connector": null,
+                    "xyz.openbmc_project.Inventory.Connector.Port": null,
+                    "com.ibm.ipzvpd.Location": {
+                        "LocationCode": "Ufcs-P0-C10-T2"
+                    },
+                    "xyz.openbmc_project.Inventory.Item": {
+                        "PrettyName": "NVMe Internal Connector"
+                    }
+                }
+            },
+            {
+                "inventoryPath": "/xyz/openbmc_project/inventory/system/chassis/motherboard/pcieslot10/pcie_card10/c10_connector3",
+                "serviceName": "xyz.openbmc_project.Inventory.Manager",
+                "inherit": false,
+                "noprime": true,
+                "ccin": ["6B87"],
+                "extraInterfaces": {
+                    "xyz.openbmc_project.Inventory.Item.Connector": null,
+                    "xyz.openbmc_project.Inventory.Connector.Port": null,
+                    "com.ibm.ipzvpd.Location": {
+                        "LocationCode": "Ufcs-P0-C10-T3"
+                    },
+                    "xyz.openbmc_project.Inventory.Item": {
+                        "PrettyName": "NVMe Internal Connector"
+                    }
+                }
+            }
+        ],
+        "/sys/bus/i2c/drivers/at24/25-0053/eeprom": [
+            {
+                "inventoryPath": "/xyz/openbmc_project/inventory/system/chassis/motherboard/pcieslot6/pcie_card6",
+                "serviceName": "xyz.openbmc_project.Inventory.Manager",
+                "replaceableAtStandby": true,
+                "concurrentlyMaintainable": true,
+                "preAction": {
+                    "collection": {
+                        "gpioPresence": {
+                            "pin": "SLOT6_EXPANDER_PRSNT_N",
+                            "value": 0
+                        },
+                        "setGpio": {
+                            "pin": "SLOT6_PRSNT_EN_RSVD",
+                            "value": 1
+                        },
+                        "systemCmd": {
+                            "cmd": "echo 25-0053 > /sys/bus/i2c/drivers/at24/bind"
```

Comment:
where did you get this value from ?

Reply:
`50001002.json` in the `master` branch had a `devAddress` tag which was used to bind the driver.
We no longer use `devAddress` tag in the `P11_Dev` JSONs and instead hard code the address in the `cmd`

https://github.com/ibm-openbmc/openpower-vpd-parser/blob/e0e1549e82c4691ad41703cfb6563c77b76cf2c4/config/ibm/50001002.json#L1290

Reply:
<img width="1129" alt="Screenshot 2024-11-13 at 8 36 50 PM" src="https://github.com/user-attachments/assets/7de6364c-f7c5-4f3c-a9ed-10d33c634d9e">
Workbook says "No card currently exists" for PCIe C6, also there is no "green box" for VPD line for C6. Should we ignore Slot C6 then?

As for the address calculation, I will take the example of PCIe C7, 
as per Workbook, C6, C7, C9 and C8 are on `I2C7` of AST2600.

Looking at the .dts file, we need to look at `&i2c6` on the .dts , since BMC FW starts at I2C0 
https://github.com/openbmc/linux/blob/5f8b7d4b2e9604d03ae06f1a2dd5a1f34c33e533/arch/arm/boot/dts/aspeed/aspeed-bmc-ibm-rainier.dts#L510

.dts tells us `i2c6` has 3 TI TMP275 temperature sensors and a NXP PCA9546 (4-channel I2C-bus switch with reset)

1. From workbook, "VPD Green box",  C7 VPD EEPROM is on `A4` `A4 >> 1 = 0x52` (Right shift because BMC FW uses 7 bit addressing)

2. If we look for `0x52` under `i2c6` in the .dts, C7 is on `i2c6mux0chn1` whose register address is `0x4D` 
`0x4D >> 1 = 0x26`

So for accessing C7's EEPROM via Linux user space, we need to bind the device at `26-0052` to the `at24` driver, so we are doing `echo 26-0052 > /sys/bus/i2c/drivers/at24/bind`